specs/online_order_spec.rb

```diff
@@ -11,61 +20,246 @@
   describe "#initialize" do
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
-
       # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+       online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", "Paid")
+      online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
       # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", "Paid")
+      online_order.customer.customer_id.must_equal "1"
+
+
+      #code below is with .customer_id
+      # online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", "Paid")
+      # # online_order.must_respond_to :customer_id
+      # online_order.customer_id.must_equal "1"
     end
 
     it "Can access the online order status" do
       # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :paid)
+      online_order.status.must_equal :paid
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
       # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :paid)
+      online_order.total.must_equal ((5 + (5 * 0.075)).round(2) + 10)
     end
 
     it "Doesn't add a shipping fee if there are no products" do
       # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new("3", {}, "1", :paid)
+      online_order.total.must_equal 0
     end
   end
 
   describe "#add_product" do
-    it "Does not permit action for processing, shipped or completed statuses" do
+    it "Does not permit action for processing statuses" do
       # TODO: Your test code here!
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :processing)
+      proc {online_order.add_product}.must_raise ArgumentError
+      # proc {Grocery::Order.find(yay_orders.length + 1)}.must_raise ArgumentError
+    end #processing
+
+    it "Does not permit action for shipped statuses" do
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :shipped)
+      proc {online_order.add_product}.must_raise ArgumentError
+    end #shipped
+
+    it "Does not permit action for completed statuses" do
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :completed)
+      proc {online_order.add_product}.must_raise ArgumentError
+    end #completed
+
+    it "Permits action for pending satuses" do
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :pending)
+      online_order.add_product("kiwi", 5).must_equal true
     end
 
-    it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+    it "Permits action for pending satuses" do
+      online_order = Grocery::OnlineOrder.new("3", {apple: 2, pear: 3}, "1", :paid)
+      online_order.add_product("kiwi", 5).must_equal true
```

Comment:
Here your test name talks about "pending statuses", but your test sets the status to `:paid`. Copy/paste error?